agent/consul/leader_connect_ca.go

```diff
@@ -762,6 +762,17 @@ func (c *CAManager) persistNewRootAndConfig(provider ca.Provider, newActiveRoot
 	return nil
 }
 
+func shouldPersistNewRootAndConfig(newActiveRoot *structs.CARoot, oldConfig, newConfig *structs.CAConfiguration) bool {
+	if newActiveRoot != nil {
+		return true
+	}
+
+	if newConfig == nil {
+		return false
+	}
```

Comment:
Just calling this out for anyone else reviewing or reading over this PR.

This condition is the new one that fixes the problem. Previously we were checking if `config != nil` to prevent reading a field from a nil. Here we instead say that if the config is nil then we know there is nothing to update, otherwise use the existing logic to see if the config has changed.